packages/compiler/src/diagnostic.ts

```diff
@@ -0,0 +1,20 @@
+import { type Token } from './lexer'
+
+export type Diagnostic = {
+  message: string
+  location:{
+    line: number
+    index: number
+    start: number
+    end: number
+  }
+}
+
+export function makeLocation(token: Token){
```

Comment:
Refactor this function to an arrow function to comply with the style guide. Also, make the location type explicit in the function signature. Maybe create a `Location` type.